src/main/scala/mesosphere/marathon/core/election/ElectionService.scala

```diff
@@ -0,0 +1,30 @@
+package mesosphere.marathon.core.election
+
+import scala.concurrent.Future
+
+trait LeadershipAbdication {
+  def abdicateLeadership(error: Boolean = false): Unit
+}
+
+trait ElectionService extends LeadershipAbdication {
+  def isLeader: Boolean
+  def leader: Option[String]
+  def offerLeadership(): Unit
+}
+
+trait ElectionDelegate {
```

Comment:
The delegate is similar to a callback, kind of the primary callback. Not sure yet whether we are able to merge these two.